src/transformers/image_transforms.py

```diff
@@ -156,9 +157,17 @@ def to_pil_image(
     image = np.squeeze(image, axis=-1) if image.shape[-1] == 1 else image
 
     # PIL.Image can only store uint8 values, so we rescale the image to be between 0 and 255 if needed.
-    do_rescale = isinstance(image.flat[0], (float, np.float32, np.float64)) if do_rescale is None else do_rescale
+    if do_rescale is None:
+        do_rescale = isinstance(image.flat[0], (float, np.floating)) and not np.allclose(image, image.astype(int))
```

Comment:
Personally, I would prefer to:
  - check if `image` in the range `[0, 1]`
    - yes: rescale
    - no: check if `np.allclose(image, image.astype(int))`
        - yes: not rescale
        - no: raise value error

But I am OK with this current version - I just feel it's a bit difficult to reason the logic (but not really from this PR)

Reply:
Agreed - that's tidier. I'll make this change 👍